packages/react-spinbutton/src/components/SpinButton/SpinButton.types.ts

```diff
@@ -1,20 +1,163 @@
+   */
+  prevFormattedValue: string;
 };
 
+export type SpinButtonFormatter = (value: number) => string;
+export type SpinButtonParser = (formattedValue: string) => number;
+export type SpinButtonKeyboardDirection = 'none' | 'up' | 'down';
+
 export type SpinButtonCommons = {
-  // TODO Add things shared between props and state here
+  /**
+   * Initial value of the control (assumed to be valid). Updates to this prop will not be respected.
+   *
+   * Use this if you intend for the SpinButton to be an uncontrolled component which maintains its
+   * own value. For a controlled component, use `value` instead. (Mutually exclusive with `value`.)
+   * @defaultvalue 0
+   */
+  defaultValue: number; // Initial value of the
+
+  /**
+   * Current value of the control (assumed to be valid).
+   *
+   * Only provide this if the SpinButton is a controlled component where you are maintaining its
+   * current state and passing updates based on change events; otherwise, use the `defaultValue`
+   * property. (Mutually exclusive with `defaultValue`.)
+   */
+  value: number;
+
+  /**
+   * Min value of the control. If not provided, the control has no minimum value.
+   */
+  min: number;
+
+  /**
+   * Max value of the control. If not provided, the control has no maximum value.
+   */
+  max: number;
+
+  /**
+   * Difference between two adjacent values of the control.
+   * This value is used to calculate the precision of the input if no `precision` is given.
+   * The precision calculated this way will always be \>= 0.
+   * @defaultvalue 1
+   */
+  step: number;
+
+  /**
+   * Custom className for the control.
+   */
+  className: string;
+
+  /**
+   * Descriptive label for the control.
+   */
+  label: string;
+
+  /**
+   * Function used to format the displayed value in the component.
+   * This allows for things like:
+   * - Displaying the value as a monetary value: $1.00
+   * - Displaying the value with a suffix: 12pt
+   *
+   * If this function is not supplied the default converts `value` to a string.
+   */
+  formatter: SpinButtonFormatter;
+
+  /**
+   * Function used to parse a formatted value back into a number.
+   * This works in conjunction with `formatter` and is its inverse (i.e., `formatter` turns
+   * 1 to $1.00 and `parser` turns $1.00 to 1).
+   *
+   * If this function is not supplied the default calls `parseFloat()` on the formatted value.
+   */
+  parser: SpinButtonParser;
+
+  /**
+   * Callback for when the committed/validated value changes. This is called _after_ `onValidate`.
+   * - User presses the up/down buttons (on single press or every spin)
+   * - User presses the up/down arrow keys (on single press or every spin)
+   * - User *commits* edits to the input text by focusing away (blurring) or pressing enter.
+   *   Note that this is NOT called for every key press while the user is editing.
+   */
+  onChange: (event: React.SyntheticEvent<HTMLElement>, newValue: SpinButtonChangeData) => void;
```

Comment:
```suggestion
  onChange: (event: React.SyntheticEvent<HTMLElement>, data: SpinButtonChangeData) => void;
```

nit: this should be `data`